### PR TITLE
tests: net: trickle move to new ztest API

### PR DESCRIPTION
**Pull request by hunterhu3000**
Move trickle tests to use new ztest API

Signed-off-by: Hu Zhenyu <zhenyu.hu@intel.com>

### CODE REVIEW

File: tests/net/trickle/src/main.c [hunk 1]
```diff
@@ -198,22 +198,21 @@ static void test_init(void)
...
-			ztest_unit_test(test_trickle_2_wait),
-			ztest_unit_test(test_trickle_1_update),
-			ztest_unit_test(test_trickle_2_inc),
-			ztest_unit_test(test_trickle_1_status),
-			ztest_unit_test(test_trickle_1_wait_long),
-			ztest_unit_test(test_trickle_stop),
-			ztest_unit_test(test_trickle_1_stopped));
-	ztest_run_test_suite(test_tickle);
-}
+ZTEST(tickle_test_suite, test_tickle)
```
Comment (nashif) [inline tests/net/trickle/src/main.c:201]: trickle, not tickle.

**Review by nashif — CHANGES_REQUESTED** [review]

File: tests/net/trickle/src/main.c (hunk 1 above)
Comment (hunterhu3000) [reply tests/net/trickle/src/main.c:201]: OK

**Review by rlubos — DISMISSED** [review]

File: tests/net/trickle/src/main.c [hunk 2]
```diff
@@ -198,22 +198,21 @@ static void test_init(void)
...
-			ztest_unit_test(test_trickle_2_wait),
-			ztest_unit_test(test_trickle_1_update),
-			ztest_unit_test(test_trickle_2_inc),
-			ztest_unit_test(test_trickle_1_status),
-			ztest_unit_test(test_trickle_1_wait_long),
-			ztest_unit_test(test_trickle_stop),
-			ztest_unit_test(test_trickle_1_stopped));
-	ztest_run_test_suite(test_tickle);
-}
+ZTEST(trickle_test_suite, test_trickle)
```
Comment (nashif) [inline tests/net/trickle/src/main.c:201]: no need for `_test_suite`

Comment (hunterhu3000) [reply tests/net/trickle/src/main.c]: Updated. Thanks to review.

**Comment by hunterhu3000** [comment]
This test case cannot be run with CONFIG_ZTEST_SHUFFLE =y, as the test case cannot be run twice.

**Review by rlubos — DISMISSED** [review]

**Review by rlubos — DISMISSED** [review]

**Comment by hunterhu3000** [comment]
All these tests need to be run by sequence, so it is not able to split.
@nashif  thanks to review and +1, so that this PR can be merged into mainline

**Review by enjiamai — DISMISSED** [review]
If these tests are strictly order, how about put the test_init/stop to setup/teardown, just a suggestion.

**Comment by hunterhu3000** [comment]
@nashif can we merge this PR?

**Review by nashif — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Pull request merged**